src/js/components/uploadDetachedFiles/UploadDetachedFilesPage.jsx

```diff
@@ -27,24 +29,47 @@ export default class UploadDetachedFilesPage extends React.Component {
 	}
 
 	componentDidMount(){
+		let showMeta = true;
 		if(this.props.params.submissionID){
```

Comment:
ComponentWillReceiveProps is only fired off when the props change. ComponentDidMount is necessary for the first setup.